### PR TITLE
Implementing file storage via Amazon S3

### PR DESCRIPTION
**Pull request by Konstantsiy**

### CODE REVIEW

File: internal/app/app.go [hunk 1]
```diff
@@ -21,10 +22,20 @@ func Start() error {
...
+	st := storage.NewStorage(storage.S3Config{
+		Region:          conf.Region,
+		AccessKeyID:     conf.AccessKeyID,
+		SecretAccessKey: conf.SecretAccessKey,
+		BucketName:      conf.BucketName,
+	})
+	err = st.InitS3ServiceClient()
+	if err != nil {
+		return err
+	}
```
Comment (papahmsolo) [inline internal/app/app.go:36]: Wrap error with some context

File: internal/storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	s3conf *S3Config
+}
+
+// NewStorage creates new file storage with the given S3 configs and bucket name.
+func NewStorage(s3conf S3Config) *Storage {
+	return &Storage{
+		svc:    &s3.S3{},
+		s3conf: &s3conf,
+	}
+}
```
Comment (papahmsolo) [inline internal/storage/storage.go:52]: - validate config values
- create session/client

File: internal/storage/storage.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	if err != nil {
+		return nil, &S3Error{fmt.Sprintf("can't create session, %v", err)}
+	}
+	return s3session, nil
+}
+
+// UploadFile uploads the given file to the bucket and returns the generated file location.
+func (s *Storage) UploadFile(file io.ReadSeeker) (string, error) {
+	secretFileKey := uuid.NewV4()
+	fileLocation := fmt.Sprintf(LocationTemplate, s.s3conf.Region, s.s3conf.BucketName, secretFileKey.String())
```
Comment (papahmsolo) [inline internal/storage/storage.go:80]: get image id as a param

**Review by papahmsolo — CHANGES_REQUESTED** [review]
👍

File: internal/storage/storage.go (hunk 3 above)
Comment (Konstantsiy) [reply internal/storage/storage.go:80]: > get image id as a param

ok, we'll have to simplify it

**Review by papahmsolo — COMMENTED** [review]
👍

**Review by papahmsolo — APPROVED** [review]

**Pull request merged**